File: code/R/S11_docking.R
# =============================================================================
# S11_docking.R
# Supplementary Figure S11 — Nafamostat binding interactions (C-panel fixed)
# *Gut* style: strong binders grouped, weak/non-specific grouped, no overlap
# Run: Rscript code/R/S11_docking.R
# =============================================================================

suppressPackageStartupMessages(library(ggplot2))

out_dir <- "figures"
dir.create(out_dir, recursive = TRUE, showWarnings = FALSE)

# ---- Binding free energy data (kcal/mol) -----------------------------------
df <- data.frame(
  target = factor(c("TNFSF10","SGS","TYRO3","TIRAP","TNKS4","TRAF2"),
                  levels = c("TNFSF10","SGS","TYRO3","TIRAP","TNKS4","TRAF2")),
  dG      = c(1.2, 1.5, 8.9, 9.7, 9.9, 9.6),
  group   = c("weak","weak","strong","strong","strong","strong")
)

# reorder: weak first (bottom), strong on top
df$target <- factor(df$target, levels = c("TNFSF10","SGS","TYRO3","TRAF2","TIRAP","TNKS4"))

# colour mapping
col_map <- c(weak = "#BDC3C7", strong = c(TYRO3 = "#8E44AD", TRAF2 = "#2980B9",
                                            TIRAP = "#27AE60", TNKS4 = "#C0392B"))

p <- ggplot(df, aes(x = dG, y = target, fill = group)) +
  geom_bar(stat = "identity", width = 0.55, colour = "#2C3E50", size = 0.3) +
  scale_fill_manual(values = c("weak" = "#BDC3C7",
                                "strong_TYRO3" = "#8E44AD",
                                "strong_TRAF2" = "#2980B9",
                                "strong_TIRAP" = "#27AE60",
                                "strong_TNKS4" = "#C0392B")) +
  geom_text(aes(label = sprintf("%.1f", dG)), hjust = -0.3, size = 3.5, colour = "#2C3E50") +
  geom_vline(xintercept = 7, colour = "#E74C3C", linetype = "dashed", lwd = 0.8) +
  annotate("text", x = 7.5, y = 6.3, label = "strong binders →", colour = "#E74C3C",
           size = 3.2, hjust = 0) +
  # annotation for C1r/C1s related
  annotate("text", x = 9.8, y = 3.5, label = "C1r/C1s related", colour = "#2980B9",
           size = 3, fontface = "italic") +
  annotate("text", x = 9.8, y = 2.5, label = "Weak/non-specific", colour = "#7F8C8D",
           size = 3, fontface = "italic") +
  scale_x_continuous(limits = c(0, 12), expand = c(0.02, 0.02),
                     name = expression(paste("Binding free energy  ", Delta, "G (kcal/mol)"))) +
  labs(y = "", title = "(C) Nafamostat binding interactions") +
  theme_bw(base_size = 10) +
  theme(plot.title = element_text(face = "bold", size = 11),
        legend.position = "none",
        axis.text.y = element_text(size = 9, colour = "#2C3E50"),
        panel.grid.minor = element_blank())

ggsave(
  filename = file.path(out_dir, "S11_docking.png"),
  plot = p, width = 9, height = 3.5, dpi = 220, bg = "white"
)
message("[S11] saved → figures/S11_docking.png")

File: code/python/S11_docking.py
"""
S11_docking.py
Supplementary Figure S11 — Nafamostat binding energies (C-panel fixed)
*Gut* style: weak/strong groups, x-axis 0–12, no overlap

Run: python code/python/S11_docking.py
"""
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import rcParams

rcParams.update({"font.family": "DejaVu Sans", "font.size": 10})
OUT = "figures"; import os; os.makedirs(OUT, exist_ok=True)

targets = ["TNFSF10","SGS","TYRO3","TRAF2","TIRAP","TNKS4"]
dG = [1.2, 1.5, 8.9, 9.6, 9.7, 9.9]
group = ["weak","weak","strong","strong","strong","strong"]
cols  = {"weak":"#BDC3C7","TYRO3":"#8E44AD","TRAF2":"#2980B9",
         "TIRAP":"#27AE60","TNKS4":"#C0392B"}

fig, ax = plt.subplots(figsize=(9, 3.2))
y_pos = list(range(6))
for i, (t, v, g) in enumerate(zip(targets, dG, group)):
    c = cols[g] if g=="weak" else cols[t]
    ax.barh(i, v, color=c, edgecolor="#2C3E50", lw=0.3, height=0.55)
    ax.text(v + 0.25, i, f"{v:.1f}", va="center", fontsize=9, color="#2C3E50")

ax.axvline(7, ls="--", color="#E74C3C", lw=0.8)
ax.text(7.3, 5.6, "strong binders →", color="#E74C3C", fontsize=9)
ax.text(9.7, 2.0, "C1r/C1s related", color="#2980B9", fontsize=8, style="italic")
ax.text(9.7, 1.0, "Weak/non-specific", color="#7F8C8D", fontsize=8, style="italic")
ax.set(xlim=(0,12), xlabel=r"Binding free energy  $\Delta$G (kcal/mol)",
       yticks=y_pos, yticklabels=targets[::-1],
       title="(C) Nafamostat binding interactions")
ax.invert_yaxis()
plt.tight_layout()
plt.savefig(f"{OUT}/S11_docking.png", dpi=220, bbox_inches="tight", facecolor="white")
print("[S11] saved → figures/S11_docking.png")
